src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -125,6 +125,54 @@ def __setattr__(self, name, value):
         self.type.settings[name] = value
 
 
+def handle_long_running_operation_exception(ex):
+    telemetry.set_exception(
+        ex,
+        fault_type='failed-long-running-operation',
+        summary='Unexpected client exception in {}.'.format(LongRunningOperation.__name__))
+    message = getattr(ex, 'message', ex)
+    error_message = 'Deployment failed.'
+
+    try:
+        correlation_id = json.loads(ex.response.content.decode())['properties']['correlationId']
+        error_message = '{} Correlation ID: {}.'.format(error_message, correlation_id)
+    except:  # pylint: disable=bare-except
+        pass
+
+    try:
+        tracking_id = re.match(r".*(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})", str(message)).group(1)
+        error_message = '{} Tracking ID: {}.'.format(error_message, tracking_id)
+    except:  # pylint: disable=bare-except
+        pass
+
+    try:
+        inner_message = json.loads(ex.response.text)['error']['details'][0]['message']
+        error_message = '{} {}'.format(error_message, inner_message)
+    except:  # pylint: disable=bare-except
+        error_message = '{} {}'.format(error_message, message)
+
+    cli_error = CLIError(error_message)
+    # capture response for downstream commands (webapp) to dig out more details
+    setattr(cli_error, 'response', getattr(ex, 'response', None))
+    raise cli_error
+
+
+def deployment_validate_table_format(result):
+    if result.get('error', None):
+        error_result = OrderedDict()
+        error_result['result'] = result['error']['code']
+        tracking_id = re.match(r".*(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})", str(result['error']['message'])).group(1)
```

Comment:
You might need to wrap it inside a try-catch block like you did for `handle_long_running_operation_exception`; otherwise the whole command might crash for unexpected error responses